core/src/main/kotlin/Formats/StructuredFormatService.kt

```diff
@@ -55,6 +56,29 @@ abstract class StructuredFormatService(locationService: LocationService,
         return StringBuilder().apply { formatText(location, content, this, listKind) }.toString()
     }
 
+    fun ContentNode.ignoreParagraphAtTop(): ContentNode {
```

Comment:
I'll look at that, previously everything was getting extra paragraphs that shouldn't causing line breaks which broke the markdown and made it invalid.